metrics-httpclient/src/main/java/com/codahale/metrics/httpclient/InstrumentedHttpClientConnectionManager.java

```diff
@@ -30,80 +30,48 @@
+                .register("https", SSLConnectionSocketFactory.getSocketFactory())
+                .build();
     }
 
     private final MetricRegistry metricsRegistry;
     private final String name;
 
-    public InstrumentedHttpClientConnectionManager(MetricRegistry metricRegistry) {
-        this(metricRegistry, getDefaultRegistry());
-    }
-
-    public InstrumentedHttpClientConnectionManager(MetricRegistry metricsRegistry,
-                                                   Registry<ConnectionSocketFactory> socketFactoryRegistry) {
-        this(metricsRegistry, socketFactoryRegistry, -1, TimeUnit.MILLISECONDS);
-    }
-
-
-    public InstrumentedHttpClientConnectionManager(MetricRegistry metricsRegistry,
-                                                   Registry<ConnectionSocketFactory> socketFactoryRegistry,
-                                                   long connTTL,
-                                                   TimeUnit connTTLTimeUnit) {
-        this(metricsRegistry, socketFactoryRegistry, null, null, SystemDefaultDnsResolver.INSTANCE, connTTL, connTTLTimeUnit, null);
-    }
-
-
-    public InstrumentedHttpClientConnectionManager(MetricRegistry metricsRegistry,
-                                                   Registry<ConnectionSocketFactory> socketFactoryRegistry,
-                                                   HttpConnectionFactory<HttpRoute, ManagedHttpClientConnection>
-                                                           connFactory,
-                                                   SchemePortResolver schemePortResolver,
-                                                   DnsResolver dnsResolver,
-                                                   long connTTL,
-                                                   TimeUnit connTTLTimeUnit,
-                                                   String name) {
-        this(metricsRegistry,
-             new DefaultHttpClientConnectionOperator(socketFactoryRegistry, schemePortResolver, dnsResolver),
-             connFactory,
-             connTTL,
-             connTTLTimeUnit,
-             name);
-    }
-
-    public InstrumentedHttpClientConnectionManager(MetricRegistry metricsRegistry,
+    private InstrumentedHttpClientConnectionManager(MetricRegistry metricsRegistry,
                                                    HttpClientConnectionOperator httpClientConnectionOperator,
                                                    HttpConnectionFactory<HttpRoute, ManagedHttpClientConnection>
                                                            connFactory,
                                                    long connTTL,
                                                    TimeUnit connTTLTimeUnit,
                                                    String name) {
         super(httpClientConnectionOperator, connFactory, connTTL, connTTLTimeUnit);
+        if (metricsRegistry==null) {
```

Comment:
This is a good defensive pattern, but across the codebase we use it. It's expected that the user will provide a not-null `MetricsRegistry`. Otherwise, one can ask why do we check the metrics registry for nullability, but not `httpClientConnectionOperator` or `connTTLTimeUnit`? I think in the long run a better approach for `Metrics` would be to use `Nullable` annotations from the `checker` framework for marking nullable parameters, but it's out of scope of this PR.

Reply:
ok, will remove the check.